src/headless/utils/parse-helpers.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * @copyright 2020, the Converse.js contributors
+ * @license Mozilla Public License (MPLv2)
+ * @description Pure functions to help funcitonally parse messages.
+ * @todo Other parsing helpers can be made more abstract and placed here.
+ */
+const helpers = {};
+
+    // Captures all mentions, but includes a space before the @
+helpers.mention_regex = /\s([@][\w_-]+)|^([@][\w_-]+)/ig;
+
+helpers.mapCleanReferences = (reference) => {
+    /* eslint-disable */
+    const { mentioned_as, ...rest } = reference;
+    return { ...rest };
+};
+
+helpers.mapAddCoordsToReferences = (indexes) => (reference, index) => {
+    const begin = indexes[index] - index;
+    return {
+        ...reference,
+        begin,
+        end: begin + reference.value.length
+    }
+};
+
+helpers.mapAddUriToReferences = (makeUriFromReference) => (reference) => ({
+    ...reference,
+    uri: makeUriFromReference(reference)
+})
+
+helpers.mapFormatMentions = mention => mention.slice(1).toLowerCase();
+
+// One of the two possible matches for `mention_regex`
+// One will always be undefined
+helpers.mapMentionsInMatches = match => match[1] || match[0];
+
+helpers.mapMentionsToTempReferences = (mention) => ({
+    mentioned_as: mention,
+    type: 'mention'
+});
```

Comment:
Looks like most of the `map*` methods take a single item as parameter and not an array, so I think it's misleading to name the helper in the plural form.

I think `mapMentionToTempReference` would therefore be better (same applies to some other map methods).